app/models/contact.rb

```diff
@@ -1,5 +1,13 @@
 class Contact < ApplicationRecord
-  validates :first_name, :last_name, presence: true
-  
   belongs_to :user
+  belongs_to :company, optional: true
+
+  VALID_PHONE_REGEX = /\A\d{3}\-\d{3}-\d{4}\z/
+  VALID_EMAIL_REGEX = /\A[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}\z/
+
+  validates :first_name, presence: {message: "can't be blank"} 
+  validates :last_name, presence: {message: "can't be blank"}
+  validates :first_name, uniqueness: { scope: [:last_name, :user_id], message: "and Last name already exist for this user" }
```

Comment:
Do we want to validate uniqueness for first and last names? Seems useful for organization/preventing duplicates in the database until the user runs into a contact with a common name, especially if app use increases.

Reply:
That’s a valid point, Joe, and that’s why we included this validation—to ensure users can’t save duplicate names, which is important for keeping their data organized. Since first and last name are the only required fields to save a contact, I relied on them to establish uniqueness for now. This approach serves as an MVP solution to prevent a single user from creating duplicate entries with the same name, while still allowing other users to save contacts with identical names. As you mentioned, this could become a challenge with common names as the app scales, so I’m open to exploring alternative approaches or adding fields to refine the uniqueness check in the future.